### PR TITLE
Simplify screenshoting logic

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This also makes the quality of screenshots configurable via appium settings interface

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@KazuCocoa  FYI

**Review by imurchie — APPROVED** [review]

**Review by jlipps — APPROVED** [review]

**Review by KazuCocoa — APPROVED** [review]
👍

File: WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m [hunk 1]
```diff
@@ -105,27 +105,12 @@ - (BOOL)fb_unlockScreen:(NSError **)error
...
-  CGRect screenRect = CGRectMake(0, 0, screenSize.width, screenSize.height);
-  // https://developer.apple.com/documentation/xctest/xctimagequality?language=objc
-  // Select lower quality, since XCTest crashes randomly if the maximum quality (zero value) is selected
-  // and the resulting screenshot does not fit the memory buffer preallocated for it by the operating system
-  NSData *imageData = [self fb_rawScreenshotWithQuality:1 rect:screenRect error:error];
-  if (nil == imageData) {
-    return nil;
-  }
-  return FBAdjustScreenshotOrientationForApplication(imageData, orientation);
+  return [self fb_rawScreenshotWithQuality:FBConfiguration.screenshotQuality rect:CGRectNull error:error];
```
Comment (KazuCocoa) [inline WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m:108]: ah, `FBAdjustScreenshotOrientationForApplication` returns ` return (NSData *)UIImagePNGRepresentation(fixedImage);`

Comment (KazuCocoa) [reply WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m:108]: On my local, when i change here to:

```
- (NSData *)fb_screenshotWithError:(NSError*__autoreleasing*)error
{
  FBApplication *activeApplication = FBApplication.fb_activeApplication;
  UIInterfaceOrientation orientation = activeApplication.interfaceOrientation;
  NSData *imageData = [self fb_rawScreenshotWithQuality:FBConfiguration.screenshotQuality rect:CGRectNull error:error];
  if (nil == imageData) {
    return nil;
  }
  return FBAdjustScreenshotOrientationForApplication(imageData, orientation);
}
```

Then,

```ruby
@@driver.update_settings({ screenshotQuality: 0 })
@@driver.save_screenshot higher_image_path #=> the file size was 181KB
```

```ruby
@@driver.update_settings({ screenshotQuality: 2 })
@@driver.save_screenshot higher_image_path #=> the file size was 248KB
```

The difference is the later was converted from JPEG to PNG. So, the size probably increased.
is there a good way to reduce the size keep png?

Comment (mykola-mokhnach) [reply WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m:108]: I don't think so, since PNG is lossless

Comment (KazuCocoa) [reply WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m:108]: yeah..

**Pull request merged**